frontend/app/routes/create.$type.tsx

```diff
@@ -298,87 +415,113 @@ export async function action({ request, params }: ActionFunctionArgs) {
   const formData = Object.fromEntries(await request.formData())
   const intent = formData?.intent
 
+  const theCookie = request.headers.get('Cookie')
+  const session = await getSession(theCookie)
+  session.set('last-action', intent as string)
+
   let apiResponse: ApiResponse = { isFailure: true, newversion: false }
-  let displayScript: boolean = false
+  const displayScript: boolean = false
+  const grantRequired: string = ''
+  let validForWallet: string
   const errors: ElementErrors = {
     fieldErrors: {},
     message: []
   }
 
-  if (intent == 'import') {
-    const result = await walletSchema.safeParseAsync(formData)
+  const actionResponse = {
+    errors,
+    apiResponse,
+    displayScript,
+    intent,
+    grantRequired
+  }
 
-    if (!result.success) {
-      errors.fieldErrors = result.error.flatten().fieldErrors
-      return json(
-        { errors, apiResponse, displayScript, intent },
-        { status: 400 }
-      )
+  // validate form data
+  const { result, payload } = await validateForm(formData, elementType)
+  if (!result.success || !payload) {
+    errors.fieldErrors = result.error?.flatten().fieldErrors || {
+      walletAddress: undefined
     }
+    return json(actionResponse, { status: 400 })
+  }
 
-    const payload = result.data
-    apiResponse = await ApiClient.getUserConfig(payload.walletAddress)
+  // no need when importing
+  if (intent != 'import') {
+    const ownerWalletAddress: string = payload.walletAddress as string
+    validForWallet = session.get('validForWallet')
+
+    if (!validForWallet || validForWallet !== ownerWalletAddress) {
+      try {
+        const walletAddress = await getValidWalletAddress(ownerWalletAddress)
+        session.set('wallet-address', walletAddress)
+
+        const quote = await fetchQuote({
+          senderAddress: walletAddress,
+          receiverAddress: walletAddress,
+          amount: 1, // 0 value fails and fee is calculated from this
+          note: 'Publisher Tools wallet verification'
+        })
```

Comment:
Do we need a quote? Can we not create a grant with an implicit quote (pass `debitAmount` instead of `quote`)? One less request from our server.